src/System.Data.SqlClient/src/System/Data/SqlClient/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,979 @@
+        //  (32766 is actually Int16.MaxValue - 1, while (-1) is for NULL termination)
+        // We must check for the lowest value between the the two
+        private const int MaxWin32PathLength = short.MaxValue - 1;
+
+        [Conditional("DEBUG")]
+        static private void AssertPathFormat(string path)
+        {
+            Debug.Assert(path != null);
+            Debug.Assert(path == path.Trim());
+            Debug.Assert(path.Length > 0);
+            Debug.Assert(path.Length <= MaxWin32PathLength);
+            Debug.Assert(path.IndexOfAny(s_invalidPathChars) < 0);
+            Debug.Assert(path.StartsWith(@"\\", StringComparison.OrdinalIgnoreCase));
+            Debug.Assert(!path.StartsWith(@"\\.\", StringComparison.Ordinal));
+        }
+
+        // SQLBUVSTS01 bugs 192677 and 193221: we cannot use System.IO.Path.GetFullPath for two reasons:
+        // * it requires PathDiscovery permissions, which is unnecessary for SqlFileStream since we are dealing with network path
+        // * it is limited to 260 length while in our case file path can be much longer
+        // To overcome the above limitations we decided to use GetFullPathName function from kernel32.dll
+        [ResourceExposure(ResourceScope.Machine)]
+        [ResourceConsumption(ResourceScope.Machine)]
+        static private string GetFullPathInternal(string path)
+        {
+            //-----------------------------------------------------------------
+            // precondition validation should be validated by callers of this method
+            // NOTE: if this method moves elsewhere, this assert should become an actual runtime check
+            // as the implicit assumptions here cannot be relied upon in an inter-class context
+            Debug.Assert(path != null);
+
+            // remove leading and trailing whitespace
+            path = path.Trim();
+            if (path.Length == 0)
+            {
+                throw ADP.Argument(SR.GetString(SR.SqlFileStream_InvalidPath), "path");
+            }
+
+            // check for the path length before we normalize it with GetFullPathName
+            if (path.Length > MaxWin32PathLength)
+            {
+                // cannot use PathTooLongException here since our length limit is 32K while
+                // PathTooLongException error message states that the path should be limited to 260
+                throw ADP.Argument(SR.GetString(SR.SqlFileStream_InvalidPath), "path");
+            }
+
+            // GetFullPathName does not check for invalid characters so we still have to validate them before
+            if (path.IndexOfAny(s_invalidPathChars) >= 0)
+            {
+                throw ADP.Argument(SR.GetString(SR.SqlFileStream_InvalidPath), "path");
+            }
+
+            // make sure path is a UNC path
+            if (!path.StartsWith(@"\\", StringComparison.OrdinalIgnoreCase))
```

Comment:
The right thing to check here is `!(PathInternal.IsDeviceUnc(path) || (path.StartsWith(@"\\") && !PathInternal.IsDevice(path)))`

Reply:
You can even add a helper to PathInternal for IsUnc() with this logic if you want.

Reply:
@JeremyKuhne 
Adding helper to PathInternal requires including file `$(CommonPath)\CoreLib\System\IO\PathInternal.Windows.cs`
This file gives a compilation error `cannot convert from 'string' to 'System.ReadOnlySpan<char>'` at the line below:
> https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/IO/PathInternal.Windows.cs#L133

To resolve the error I can explicitly convert the `string` to `span`, any suggestions/comments ?
`if (IsPartiallyQualified(path.AsSpan()) || IsDevice(path.AsSpan()))`

Reply:
> You can even add a helper to PathInternal for IsUnc() with this logic if you want.

Did you mean the method [IsDeviceUNC()](https://github.com/dotnet/corefx/blob/1b9cf748834740ad843828384cfd6967e991b115/src/Common/src/CoreLib/System/IO/PathInternal.Windows.cs#L164) ?